packages/amplify-category-hosting/__tests__/lib/S3AndCloudFront/configuration-manager.test.js

```diff
@@ -0,0 +1,108 @@
+                "StackName": "checkhosting-20190226163640",
+                "StackId": "arn:aws:cloudformation:us-west-2:mockAccountId:stack/checkhosting-20190226163640/2c061610-3a28-11e9-acf3-02ee71065ed8"
+            }
+        },
+        "hosting": {
+            "S3AndCloudFront": {
+                "service": "S3AndCloudFront",
+                "providerPlugin": "awscloudformation",
+                "providerMetadata": {
+                    "s3TemplateURL": "https://s3.amazonaws.com/checkhosting-20190226163640-deployment/amplify-cfn-templates/hosting/template.json",
+                    "logicalId": "hostingS3AndCloudFront"
+                },
+                "lastPushTimeStamp": "2019-02-27T00:39:17.966Z",
+                "output": {
+                    "S3BucketSecureURL": "https://checkosting-20190226163802-hostingbucket-dev.s3.amazonaws.com",
+                    "WebsiteURL": "http://checkosting-20190226163802-hostingbucket-dev.s3-website-us-west-2.amazonaws.com",
+                    "Region": "us-west-2",
+                    "HostingBucketName": "checkosting-20190226163802-hostingbucket-dev"
+                },
+                "lastPushDirHash": "83Bhmmec48dILMj3mi2T25B4700="
+            }
+        }
+    }
+
+    const mockProjectConfig = {
+        "projectName": "mockProjectName",
+        "version": "2.0",
+        "frontend": "javascript",
+        "javascript": {
+            "framework": "none",
+            "config": {
+                "SourceDir": "src",
+                "DistributionDir": "dist",
+                "BuildCommand": "npm run-script build",
+                "StartCommand": "npm run-script start"
+            }
+        },
+        "providers": [
+            "awscloudformation"
+        ]
+    };
+
+    const mockContext = {
+        exeInfo: {
+            parameters: {},
+            projectConfig: mockProjectConfig,
+            serviceMeta: mockAmplifyMeta.hosting.S3AndCloudFront
+        }
+    }; 
+
+    beforeAll(() => {
+        inquirer.prompt = jest.fn(); 
+        configureWebsite.configure = jest.fn(); 
+        configurePublish.configure = jest.fn(); 
+        configureCloudFront.configure = jest.fn(); 
+    }); 
+
+    beforeEach(() => {
+        inquirer.prompt.mockClear();
+        configureWebsite.configure.mockClear();
+        configureCloudFront.configure.mockClear();
+        configurePublish.configure.mockClear();
+    });
+
+    test('init', async () => {
+        inquirer.prompt.mockResolvedValueOnce({HostingBucketName: 'mockHostingBucketName'}); 
+        await configurationManager.init(mockContext); 
+        expect(mockContext.exeInfo.parameters.bucketName).toBeDefined(); 
+        expect(configureWebsite.configure).toBeCalled(); 
+        expect(configureWebsite.configure.mock.calls[0][0]).toBe(mockContext); 
+    });
+
+    test('configure', async () => {
+        inquirer.prompt.mockResolvedValueOnce({section: 'Website'}); 
```

Comment:
Nit: I would prefer if you export `configureHostingComponent` and `checkBucketName` and then use [`jest.spyOn`](https://jestjs.io/docs/en/jest-object#jestspyonobject-methodname-accesstype) on these methods.

Doing this will give you chance to test these methods with all the cases without worrying about the `configure` method as its just a composer